jmaqs-webservices-jdk11/src/main/java/com/magenic/jmaqs/webservices/jdk11/WebServiceDriver.java

```diff
@@ -18,37 +27,377 @@
+   * @param mediaType The Media Type
+   * @param content The Http Request Content
+   * @param additionalHeaders The Additional Headers for the Http Request
+   * @return The Http Request
+   */
+  protected HttpRequest buildHttpRequest(String requestUri, RequestType requestType,
+      MediaType mediaType, String content, Map<String, String> additionalHeaders) {
+    HttpRequest.Builder builder = this.baseHttpRequestBuilder.copy();
+
+    builder
+      .header("Content-Type", mediaType.toString())
+      .uri(URI.create(requestUri));
+
+    for (Map.Entry<String, String> header : additionalHeaders.entrySet()) {
+      builder.header(header.getKey(), header.getValue());
+    }
+
+    if (requestType.equals(RequestType.POST)) {
+      return builder.POST(HttpRequest.BodyPublishers.ofString(content)).build();
+    } else if (requestType.equals(RequestType.PUT)) {
+      return builder.PUT(HttpRequest.BodyPublishers.ofString(content)).build();
+    } else if (requestType.equals(RequestType.DELETE)) {
+      return builder.DELETE().build();
+    } else if (requestType.equals(RequestType.PATCH)) {
+      return builder.method("PATCH", HttpRequest.BodyPublishers.ofString(content)).build();
+    }
+    return builder.GET().build();
+
+  }
+
+  /**
+   * Deserialize the Response.
+   * @param httpResponse Http Response to deserialize
+   * @param mediaType The Media Type
+   * @param type The Object Type
+   * @param <T> The Object Type
+   * @return The De-serialized Object
+   * @throws IOException if the exception is thrown
+   */
+  private <T> T deserializeResponse(HttpResponse<String> httpResponse, MediaType mediaType, Type type)
+      throws IOException {
+    if (mediaType.equals(MediaType.APP_XML)) {
+      return WebServiceUtilities.deserializeXml(httpResponse, type);
+    } else {
+      return WebServiceUtilities.deserializeJson(httpResponse, type);
+    }
+  }
+
+  /**
+   * Ensure the HTTP response was successful, if not throw a user friendly error message.
+   * @param response The HTTP response
+   */
+  private static void ensureSuccessStatusCode(HttpResponse<String> response) {
+    // Make sure a response was returned
+    if (response == null) {
+      throw new NullPointerException(HttpStatus.NO_CONTENT.toString() + " Response was null");
+    }
+
+    // Check if it was a success and if not create a user friendly error message
+    if (response.statusCode() != HttpStatus.OK.value()) {
+      throw new NotAcceptableStatusException(
+          String.format("Response did not indicate a success. %s Response code was: %d",
+              System.lineSeparator(), response.statusCode()));
+    }
+  }
+
+  /**
+   * Ensure the HTTP response has specified status, if not throw a user friendly error message.
+   * @param response The HTTP response
+   * @param expectedStatus Assert a specific status code was returned
+   */
+  private static void ensureStatusCodesMatch(HttpResponse<String> response, HttpStatus expectedStatus) {
```

Comment:
Now that we created our own Enum, shouldn't we remove the spring dependencies?

Reply:
The HttpStatus is being used for this method. If we dont need this method, we can remove the spring dependencies, otherwise we'll need them to stay, I suppose. I'm thinking we wouldn't need this, and can clean up the ensureSuccessStatusCode method to remove those dependencies